src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/shared_form_components.tsx

```diff
@@ -0,0 +1,394 @@
+            'There is no way to get fields or functions values from an ES|QL query at the moment.',
+        })}
+        condition={isDisabled}
+        anchorProps={{
+          css: { width: '100%' },
+        }}
+      >
+        <EuiFormRow
+          label={i18n.translate('esql.flyout.controlTypeOptionsOptions.label', {
+            defaultMessage: 'Type',
+          })}
+          fullWidth
+        >
+          <EuiComboBox
+            aria-label={i18n.translate('esql.flyout.controlTypeOptionsOptions.placeholder', {
+              defaultMessage: 'Select a control type',
+            })}
+            placeholder={i18n.translate('esql.flyout.controlTypeOptionsOptions.placeholder', {
+              defaultMessage: 'Select a control type',
+            })}
+            singleSelection={{ asPlainText: true }}
+            options={controlTypeOptions}
+            selectedOptions={[controlFlyoutType]}
+            onChange={onTypeChange}
+            fullWidth
+            isDisabled={isDisabled}
+            compressed
+            data-test-subj="esqlControlTypeDropdown"
+            inputPopoverProps={{
+              'data-test-subj': 'esqlControlTypeInputPopover',
+            }}
+          />
+        </EuiFormRow>
+      </TooltipWrapper>
+      <EuiSpacer size="m" />
+    </>
+  );
+}
+
+export function VariableName({
+  variableName,
+  isControlInEditMode,
+  onVariableNameChange,
+}: {
+  variableName: string;
+  isControlInEditMode: boolean;
+  onVariableNameChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}) {
+  const genericContent = i18n.translate('esql.flyout.variableName.helpText', {
+    defaultMessage: 'This name will be prefaced with an "?" in the editor',
+  });
+  const isDisabledTooltipText = i18n.translate('esql.flyout.variableName.disabledTooltip', {
+    defaultMessage: 'You can’t edit a control name after it’s been created.',
+  });
+  return (
+    <EuiFormRow
+      label={i18n.translate('esql.flyout.variableName.label', {
+        defaultMessage: 'Name',
+      })}
+      helpText={i18n.translate('esql.flyout.variableName.helpText', {
+        defaultMessage: 'This name will be prefaced with an "?" in the editor',
+      })}
+      fullWidth
+      autoFocus
+      isInvalid={
+        !variableName ||
+        (esqlVariablesService.variableExists(variableName.replace('?', '')) && !isControlInEditMode)
+      }
+      error={
+        !variableName
+          ? i18n.translate('esql.flyout.variableName.error', {
+              defaultMessage: 'Variable name is required',
+            })
+          : esqlVariablesService.variableExists(variableName) && !isControlInEditMode
+          ? i18n.translate('esql.flyout.variableNameExists.error', {
+              defaultMessage: 'Variable name already exists',
+            })
+          : undefined
+      }
+    >
+      <EuiToolTip
+        content={
+          isControlInEditMode ? `${genericContent} ${isDisabledTooltipText}` : genericContent
```

Comment:
On edit, only show isDisabledTooltipText:

```suggestion
          isControlInEditMode ? isDisabledTooltipText : genericContent
```

Reply:
Done https://github.com/elastic/kibana/pull/202875/commits/97500e4d5225fc6f1a6d242b3dd80f2c04ac8b0a